task/sql-tasks.js

```diff
@@ -281,7 +457,28 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT DISTINCT
+            CompanyName,
+            Products.ProductName AS "ProductName",
+            t.PricePerItem2 AS "PricePerItem"
+        FROM \`Customers\`
+        INNER JOIN Orders ON Customers.CustomerID = Orders.CustomerID
+        INNER JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+        INNER JOIN Products ON Products.ProductID = OrderDetails.ProductID
+        RIGHT JOIN(
+            SELECT
+                Customers.CustomerID,
+                MAX(OrderDetails.UnitPrice) AS "PricePerItem2"
+            FROM \`Customers\`
+            INNER JOIN Orders ON Customers.CustomerID = Orders.CustomerID
+            INNER JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+            INNER JOIN Products ON Products.ProductID = OrderDetails.ProductID
+            GROUP BY Customers.CustomerID) t 

```

Comment:
You made a join with temporary table `t` here. We need to try to avoid the cases when we work with MYSQL to have better performance. We can avoid it here

Reply:
Okay. How?

Reply:
if I'll answer it's gonna be boring )
BUT I have the two "explains"
I've [compared the execution plans via DataGrip](https://www.screencast.com/t/uhI6VL7K1Rou)
You can check execution plans with [mysql workbench](https://dev.mysql.com/doc/workbench/en/wb-performance-explain.html) or some online tools.

Reply:
Well...But how to get execution plan in DG? I couldnt find it in google..